drivers/char/serial.c

```diff
@@ -695,7 +695,7 @@ void serial_init(void)
 		}
 
 		/* check if a serial tty will act as a system console */
-		for(n = 0, syscon = 0; n < NR_SERIAL; n++) {
+		for(n = 0, syscon = 0; n < NR_SYSCONSOLES; n++) {
```

Comment:
What type of warning this code does?
If you think there is a bug in this code, please, create a new PR for it, just to have a good tracking of bugs. Otherwise this fix would be hidden on a non-related PR.

Reply:
> What type of warning this code does?
> If you think there is a bug in this code

The problem is that in line 699:
```
if(is_serial(sysconsole_table[n].dev)) {
```
you are indexing `sysconsole_table[]` with NR_SERIAL, while it is declared using NR_SYSCONSOLES:
```
struct serial serial_table[NR_SERIAL];
```
Thus, if NR_SERIAL is ever > NR_SYSCONSOLE, an array over index occurs!

> If you think there is a bug in this code, please, create a new PR for it, just to have a good tracking of bugs. 

This is a minor fix since NR_SERIAL is usually 1. Not sure if a separate PR is needed. I added it here because it was caught by my newer compiler.

Reply:
> This is a minor fix since NR_SERIAL is usually 1.

Oops - NR_SERIAL is usually 5. Would you like this fix removed from this PR, and/or would you like to investigate this more thoroughly?

Reply:
> Oops - NR_SERIAL is usually 5. Would you like this fix removed from this PR, and/or would you like to investigate this more thoroughly?

Yes, please.

Reply:
Ok, removed.

Reply:
After looking at the code a bit more, I'd like your input as to whether my proposed fixed of using NR_SYSCONSOLES instead of NR_SERIAL is actually what you want. I am not able to test with multiple console so please let me know what you think.

Reply:
Here's the compiler warning:
```
x86_64-linux-musl-gcc -m32 -march=i386 -std=c89 -D__KERNEL__ -DCUSTOM_CONFIG_H  -I/Users/greg/net/Fiwix/include -O2 -fno-pie -fno-common -ffreestanding -Wall -Wstrict-prototypes  -c -o serial.o serial.c
serial.c: In function 'serial_init':
serial.c:699:36: warning: iteration 1 invokes undefined behavior [-Waggressive-loop-optimizations]
  699 |    if(is_serial(sysconsole_table[n].dev)) {
      |                 ~~~~~~~~~~~~~~~~~~~^~~~
serial.c:698:3: note: within this loop
  698 |   for(n = 0, syscon = 0; n < NR_SERIAL; n++) {
      |   ^~~
```

Reply:
> I am not able to test with multiple console so please let me know what you think.

I've just tested your patch with multiple consoles and all continue working fine, so go ahead with the patch.